### PR TITLE
Add the XHarness bootstrap script

### PR DESCRIPTION
**Pull request by premun**
Adds a simple script that installs XHarness locally in a folder so that you can try XHarness out easily.

### CODE REVIEW

**Review by mandel-macaque — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -9,14 +9,26 @@ It can locate devices/emulators, install a given application, run it and collect
...
 It can output test results in various different formats from text to xUnit/NUnit XML.
 
-## Running the tool
+## System requirements
 
 The tool requires **.NET Core 3.1.201** or later to be run. It is packaged as a `dotnet tool` command and can be installed using the [dotnet tool CLI](https://docs.microsoft.com/en-us/dotnet/core/tools/).
 
 - The iOS scenarios require you to run the tool on MacOS with full Xcode installation
-- Android scenarios are supported on Linux, macOS and Windows systems
+- Android scenarios are supported on Linux, macOS and Windows systems with Android emulators
```
Comment (akoeplinger) [inline README.md:17]: not just emulators, also devices :)

File: README.md [hunk 2]
```diff
@@ -9,14 +9,26 @@ It can locate devices/emulators, install a given application, run it and collect
...
 - Browsers scenarios are supported on Linux systems
 
+## Try the tool out quickly
+
+If you want to test the tool quickly (MacOS and Linux only), following script will install the required .NET SDK and the XHarness tool locally in the current folder.
+It will also pull a sample iOS app for you and you can then try running the XHarness tool right away (instructions will be printed on screen):
```
Comment (akoeplinger) [inline README.md:23]: I don't think we should do this in the bootstrap script. It's unrelated to xharness and will only work on macOS.

File: tools/xharness-bootstrap.sh [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+echo 'Downloading sample app...'
+curl https://netcorenativeassets.blob.core.windows.net/resource-packages/external/macos/test-ios-app/"$app_name.zip" -o "$app_name.zip"
+
+echo "Unpacking $app_name.."
+tar -xzf "$app_name.zip"
+rm "$app_name.zip"
+
+echo "Sample app ready in $here/$app_name"
+
+printf "Run XHarness using:\n\033[0;36m./xharness/xharness ios test -a=%s -o=o -t=ios-simulator-64\033[0m\n\n" "$app_name"
```
Comment (akoeplinger) [inline tools/xharness-bootstrap.sh:60]: I think we can still print this usage instruction (maybe with --long-option's instead so it's clearer what they do) but we shouldn't download the sample .app

**Review by akoeplinger — CHANGES_REQUESTED** [review]

**Review by greenEkatherine — APPROVED** [review]

**Comment by premun** [comment]
This is the output of the script:
image

File: README.md [hunk 4]
```diff
@@ -9,14 +9,26 @@ It can locate devices/emulators, install a given application, run it and collect
...
 The tool requires **.NET Core 3.1.201** or later to be run. It is packaged as a `dotnet tool` command and can be installed using the [dotnet tool CLI](https://docs.microsoft.com/en-us/dotnet/core/tools/).
 
 - The iOS scenarios require you to run the tool on MacOS with full Xcode installation
 - Android scenarios are supported on Linux, macOS and Windows systems
 - Browsers scenarios are supported on Linux systems
 
+## Try the tool out quickly
+
+If you want to test the tool quickly (MacOS and Linux only), following script will install the required .NET SDK and the XHarness tool locally in the current folder.
+It will also pull a sample iOS app for you and you can then try running the XHarness tool right away (instructions will be printed on screen):
```
Comment (akoeplinger) [inline README.md:23]: This sentence is no longer true in the latest iteration :)

**Review by akoeplinger — APPROVED** [review]

**Pull request merged**